crux_kv/src/lib.rs

```diff
@@ -87,26 +85,28 @@ impl std::fmt::Debug for KeyValueOperation {
 ///
 /// Note: we can't use `Result` and `Option` here because generics are not currently
 /// supported across the FFI boundary, when using the builtin typegen.
-#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[derive(Facet, Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[repr(C)]
 pub enum KeyValueResult {
-    Ok { response: KeyValueResponse },
-    Err { error: KeyValueError },
+    Ok(KeyValueResponse),
+    Err(KeyValueError),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Facet, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[repr(C)]
 pub enum KeyValueResponse {
     /// Response to a `KeyValueOperation::Get`,
     /// returning the value stored under the key, which may be empty
-    Get { value: Value },
+    Get(Option<Vec<u8>>),
     /// Response to a `KeyValueOperation::Set`,
     /// returning the value that was previously stored under the key, may be empty
-    Set { previous: Value },
+    Set(Option<Vec<u8>>),
     /// Response to a `KeyValueOperation::Delete`,
     /// returning the value that was previously stored under the key, may be empty
-    Delete { previous: Value },
+    Delete(Option<Vec<u8>>),
     /// Response to a `KeyValueOperation::Exists`,
     /// returning whether the key is present in the store
-    Exists { is_present: bool },
+    Exists(bool),
```

Comment:
I'm reasonably sure we're going back to what we once had, which we changed to be more explicit about what the carried values are, especially for the `previous` case. 

If these are purely a matter of taste, I think I'd leave them alone, especially since this is probably quite a heavily used type, so the breaking change is high impact.